refactor inventory filtering and sorting

ThyWoof opened the pull request:
fixes the annoying glitch that kept the remaining time gauge on non light source items.

ThyWoof commented:
![image](https://user-images.githubusercontent.com/13087646/149645732-533c67aa-07e3-42eb-b0e7-8159a4e4f8c0.png)

ThyWoof commented:
![image](https://user-images.githubusercontent.com/13087646/149645737-087f95ad-04a9-433e-9c1d-68a625f9f23c.png)
yellow icon stays with candle.

ThyWoof commented:
these are the first step to try to add a custom icon on top of scrolls that cannot be used => https://github.com/ChrisPJohn/SolastaCommunityExpansion/issues/416

ThyWoof commented:
hold merging this one...

ThyWoof commented:
![image](https://user-images.githubusercontent.com/13087646/149650821-8bbc3a45-7e30-4032-a1ec-b3f58f29fc81.png)
non-proficient items displayed with a red border

PhilPJL commented:
The red border is a great idea, but I think it's being left applied to prefabs, because

First I view this character
![image](https://user-images.githubusercontent.com/271656/149661036-29741f1a-45f1-4d76-930c-a3507a57ca1a.png)

Then this one
![image](https://user-images.githubusercontent.com/271656/149661060-6fdf286b-9c2d-4016-979b-9ed8338e0fcf.png)
None of those were red previously and shouldn't be.

PhilPJL reviewed (changes requested):
Red border issue
Reset to default order issue

PhilPJL commented:
If I choose a specific sort order, switch away and back again, the items are in the same order.
If I press 'reset' the sort order is 'default' (whatever that is - random?).  If I switch away and back the items are in a different order.

ThyWoof commented:
@PhilPJL : I gave up for now on getting slots with different colors. This fixes the reset button and a cleaner intention in code.

ThyWoof commented:
summary for this request:

1. No more glitches on inventory visual effects (i.e.: deity mark, remaining usage, etc.) although game has a bug that will loose these icons if you drag an item;
2. Removed "Requires Restart". One can even flip the Mod UI toggle with the inventory window open (although the controls might be out of sync if one does that. clicking on any control bring them back. no issues even if there are filters enabled and the setting is turned off. No items are lost);
3. Removed the "Reset" button;

ThyWoof commented:
ready for another review round @PhilPJL

PhilPJL commented on SolastaCommunityExpansion/Models/InventoryManagementContext.cs:
```diff
@@ -73,211 +108,191 @@ internal static void Load()
-                filterGuiDropdown.value = 0;
-                bySortGroup.Inverted = false;
-                sortGuiDropdown.value = 0;
+            SortCategories.ForEach(x => sortOptions.Add(new TMP_Dropdown.OptionData() { text = x }));
```
Change to use OptionDataAdvanced because of known bugs in DLC.  Can use tooltip if required.

```
            SortCategories.ForEach(x => sortOptions.Add(new OptionDataAdvanced
            {
                text = x,
                //TooltipContent = "<TODO>"
            }));

            SortGuiDropdown.AddOptions(sortOptions);
            SortGuiDropdown.template.sizeDelta = new Vector2(1f, 208f);
            //SortGuiDropdown.UpdateTooltip();
```

PhilPJL commented on SolastaCommunityExpansion/Models/InventoryManagementContext.cs:
```diff
@@ -73,211 +108,191 @@ internal static void Load()
+            FilterGuiDropdown.onValueChanged.AddListener(delegate { SelectionChanged(); });
 
-            FilterCategories.ForEach(x => filterOptions.Add(new TMP_Dropdown.OptionData() { text = x.FormatTitle() }));
+            ItemCategories.ForEach(x => filterOptions.Add(new TMP_Dropdown.OptionData() { text = x.FormatTitle() }));
```
Change to use OptionDataAdvanced because of known bugs in DLC. Can use tooltip if required.

```
            ItemCategories.ForEach(x => filterOptions.Add(new OptionDataAdvanced
            {
                text = x.FormatTitle(),
                //TooltipContent = "<TODO>"
            }));

            FilterGuiDropdown.AddOptions(filterOptions);
            FilterGuiDropdown.template.sizeDelta = new Vector2(1f, 208f);
            //FilterGuiDropdown.UpdateTooltip();

```
